File: week-01/day-01/algo-deaf-grandma/deaf_grandma.js
let goodbyeCount = 0; //Number of times "GOODBYE!" was entered
console.log("HEY KID!"); //Initial response

//Continue prompting user until "GOODBYE!" is entered twice
while (true) {
    let input = prompt(); //Get input from user

    //"WHAT?!" response if cancel button was pressed in prompt
    //or no words were entered
    if(input === null || input.trim() === "")
        console.log("WHAT?!");
    //"SPEAK UP, KID!" response if any lowercase letter found
    else if (input.search("[a-z]") != -1)
        console.log("SPEAK UP, KID!");
    //Increment goodbyeCount when "GOODBYE!" is entered
    //and print appropriate response, breaking out of loop if count is 2
    else if (input.trim() === "GOODBYE!") {
        console.log(`${goodbyeCount === 0 ? "LEAVING SO SOON?" : "LATER, SKATER!"}`);
        if(goodbyeCount != 0)
            break;
        goodbyeCount++;
    }
    //All uppercase letters or symbols were entered
    else
        console.log("NO, NOT SINCE 1946!");
}

File: week-01/day-01/algo-deaf-grandma/deaf_grandma.py
import re #Regular expressions module

print("HEY KID!") #Print initial response

goodbyeCount = 0 #Number of times "GOODBYE!" was entered

#Take user input until "GOODBYE!" is entered twice
while True:
    answer = input("> ") #Get input from user

    #User input was empty
    if answer.strip() == "":
        print("WHAT?!")
    #User entered one or more lowercase letters
    elif re.match("[a-z]", answer):
        print("SPEAK UP, KID!")
    #Increment goodbyeCount, print appropriate response,
    #and break from loop if count is 2
    elif answer.strip() == "GOODBYE!":
        print(f"{'LEAVING SO SOON?' if goodbyeCount == 0 else 'LATER, SKATER!'}")
        if goodbyeCount != 0:
            break
        goodbyeCount += 1
    #All uppercase or symbols were entered
    else:
        print("NO, NOT SINCE 1946!")
